ios/BT/API/Model/DownloadedPackage.swift

```diff
@@ -2,35 +2,44 @@ import Foundation
 import ZIPFoundation
 import CryptoKit
 
-protocol DownloadedPackage {
-  func writeSignatureEntry(toDirectory directory: URL, filename: String) throws -> URL
-  func writeKeysEntry(toDirectory directory: URL, filename: String) throws -> URL
-}
-
-struct DownloadedPackageImpl: DownloadedPackage {
+protocol DownloadableFile {
 
-  init(keysBin: Data, signature: Data) {
-    bin = keysBin
-    self.signature = signature
-  }
+  static func create(from data: Data) -> Self?
+}
 
-  init?(compressedData: Data) {
-    guard let archive = Archive(data: compressedData, accessMode: .read) else {
+class DownloadedPackage: DownloadableFile {
+  
+  static func create(from data: Data) -> Self? {
+    guard let archive = Archive(data: data, accessMode: .read) else {
       return nil
     }
     do {
-      guard let package = try archive.extractKeyPackage() as? DownloadedPackageImpl else {
-        return nil
-      }
-      self = package
+      return try archive.extractKeyPackage() as? Self
     } catch {
       return nil
     }
   }
 
+
+  init(keysBin: Data, signature: Data) {
+    bin = keysBin
+    self.signature = signature
+  }
+
   let bin: Data
   let signature: Data
 
+  func writeSignatureEntry(toDirectory directory: URL, filename: String) throws -> URL {
+    let url = directory.appendingPathComponent(filename).appendingPathExtension("sig")
```

Comment:
Not at all necessary, but could use a private extension on `String` for this: 

```
private extension String {
  static let binExtension = "bin"
  static let sigExtension = "sig"
  static let exportFilename = "export"
  static var binEntry: String {
    "\(exportFilename).\(binExtension)"
  }
  static var sigEntry: String {
    "\(exportFilename).\(sigExtension)"
  }
}
```

Reply:
will do